### PR TITLE
Unify markdown rendering #2

### PR DESCRIPTION
**Pull request by beyang**
Previously, we had copy-pasted code replicating markdown rendering in a codeintel graphql resolver and a frontend graphql resolver. We just updated our markdown rendering library to one that's actually maintained (https://github.com/sourcegraph/sourcegraph/pull/46960) , and it seems appropriate to update the codeintel graphql resolver to use it.

This changeset moves markdown rendering into a shared `internal/markdown` package and updates the codeintel and frontend graphql resolvers to use this package.

## Test plan

Tested locally

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.51% (+14.35 kb) 🔺 | 0.13% (+19.25 kb) 🔺 | 0.04% (+4.89 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/bl/SuperAuguste/update-markdown-unify-markdown/compare-report.html;tab=live_object) for a full comparison between the commits 7e7ea8f and 9449278 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Review by Numbers88s — APPROVED** [review]

File: internal/codeintel/resolvers/all.go [hunk 1]
```diff
@@ -1,18 +1,17 @@
 package resolvers
 
 import (
-	"context"
-	"regexp" //nolint:depguard // regexps are passed to bluemonday, which expects the std ones
+	"context" //nolint:depguard // regexps are passed to bluemonday, which expects the std ones
```
Comment (efritz) [inline internal/codeintel/resolvers/all.go:4]: Is this comment outdated now?

**Review by efritz — APPROVED** [review]
Just one lint comment question.

File: internal/codeintel/resolvers/all.go (hunk 1 above)
Comment (beyang) [reply internal/codeintel/resolvers/all.go]: Removed

**Pull request merged**